pkg/certificate/providers/tresor/certificate_manager.go

```diff
@@ -22,7 +22,11 @@ func (cm *CertManager) issue(cn certificate.CommonName, validityPeriod time.Dura
 		return nil, errNoIssuingCA
 	}
 
-	certPrivKey, err := rsa.GenerateKey(rand.Reader, rsaBits)
+	certKeyBitSize := rsaBits
+	if cm.cfg != nil {
+		certKeyBitSize = cm.cfg.GetCertKeyBitSize()
+	}
```

Comment:
If tests are passing `cfg` as `nil`, we should fix those instead. Non nil `cfg` should not be allowed when initializing a certificate provider, so we could drop this check.

Reply:
Not required with the new proposal: https://github.com/openservicemesh/osm/pull/3837#issuecomment-885248364